prow/test/integration/test/sinker_test.go

```diff
@@ -282,11 +282,10 @@ func TestDeletePod(t *testing.T) {
 				}
 			}
 
-			// Make sure pod is deleted, it'll take roughly 1 minutes
-			// Don't care about the outcome, will check later
-			t.Logf("Wait for sinker deleting pod or timeout in 1 minutes: %s", pod.Name)
+			// Make sure pod is deleted.
+			t.Logf("Wait for sinker deleting pod or timeout in 3 minutes: %s", pod.Name)
 			var exist bool
-			wait.Poll(time.Second, 1*time.Minute, func() (bool, error) {
+			wait.Poll(time.Second, 3*time.Minute, func() (bool, error) {
```

Comment:
Is this reasonable? If [sinker is resyncing every 5 seconds](https://github.com/kubernetes/test-infra/blob/7d65a0b2fde3dd0a2a61de8b10a000181534e67d/prow/test/integration/config/prow/config.yaml#L7) then this seems like much too long of a delay.
Perhaps we need to be waiting for sinker to reach the "Running" state before doing this test? I'm not sure what else would explain such a significant delay.

Reply:
So it turns out that we already wait for all prow components to become "ready" (https://github.com/kubernetes/test-infra/blob/64221f7c3e32afd3a7ccfafd0eb1ca4cc86a0906/prow/test/integration/lib.sh#L143 and https://github.com/kubernetes/test-infra/blob/64221f7c3e32afd3a7ccfafd0eb1ca4cc86a0906/prow/test/integration/lib.sh#L196-L202). So the only explanation is that either sinker is flaking out or taking too long (for whatever reason).

I just remembered that I updated the test logs to dump KIND logs upon a test failure. Looks like sinker was indeed in a bad state: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/test-infra/28042/pull-test-infra-integration/1593065403758678016/artifacts/kind_logs/kind-prow-integration-control-plane/pods/default_sinker-758cc76fb8-zh6p9_a83543ad-988b-4a80-9014-65e51457e444/sinker/0.log (could not list pods). So it appears that the `sinker` "Role" was not created yet. The real fix, then, is to add an additional waiting stage during Prow component setup so that we create these roles first before moving on to deploying sinker. We should probably do this for all components.